### PR TITLE
Improvement on old Container Swappable

### PR DESCRIPTION
**Pull request by FlaviooLima**
| - | - |
| --- | --- |
| Issue | --- |
| Decisions | Improvement on old Container Swappable (V2)  |
| Animated GIF | Mar-05-2020 17-02-38 |

### CODE REVIEW

**Comment by gcandal** [comment]
@FlaviooLima is the cleanup complete? Is this ready for review?

File: react/components/atoms/container-swipeable/container-swipeable.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+    Modal,
+    View,
+    TouchableOpacity,
+    PanResponder,
+    Dimensions
+} from "react-native";
 import { initialWindowSafeAreaInsets } from "react-native-safe-area-context";
 import PropTypes from "prop-types";
 
+//@TODO This method for AndroidSoftBar
```
Comment (3rdvision) [inline react/components/atoms/container-swipeable/container-swipeable.js]: still a todo?

File: react/components/atoms/container-swipeable/container-swipeable.js [hunk 2]
```diff
@@ -12,81 +24,156 @@ export class ContainerSwipeable extends PureComponent {
...
+        }
+    };
+
     toggle = () => {
         if (this.animating) {
             return;
         }
 
-        if (this.state.visible) {
+        if (this.state.itemsVisible) {
```
Comment (3rdvision) [inline react/components/atoms/container-swipeable/container-swipeable.js]: why not just 
```
this.setState( state => ({
  visible: !state.visible,
  itemsVisible: !state.itemsVisible
}));
```

File: react/components/atoms/container-swipeable/container-swipeable.js [hunk 3]
```diff
@@ -12,81 +24,156 @@ export class ContainerSwipeable extends PureComponent {
...
+                modalVisible: true
+            });
         }
     };
 
     onLayout = event => {
-        this.setState({
-            containerChildrenHeight: event.nativeEvent.layout.height
-        });
+        if (this.state.containerChildrenHeight) {
```
Comment (3rdvision) [inline react/components/atoms/container-swipeable/container-swipeable.js]: Update to oneliner for better readability ? :)
`if (this.state.containerChildrenHeight) return;`

**Comment by joamag** [comment]
@FlaviooLima @gcandal not being able to work with it on android emulator

**Comment by FlaviooLima** [comment]
@joamag this is working on android!
If you continue to having problems maybe we need to cross information about environments

**Comment by joamag** [comment]
@FlaviooLima Weird I'll try again

**Comment by gcandal** [comment]
As we discussed together, I'm closing this since the new approach required for @NFSS10 's task will most likely change too much stuff. He'll finish his task of a container that can be modal-like or be inserted in some section of the page and then use this PR as basis for adding swipe capabilities to it.

**Pull request closed without merging**